neural_network/2_hidden_layers_neural_network.py

```diff
@@ -0,0 +1,203 @@
+        # layer1 is the layer connecting the input nodes with
+        # the first hidden layer nodes
+        self.layer1 = sigmoid(numpy.dot(self.input, self.weights1))
+
+        # layer2 is the layer connecting the first hidden set of nodes
+        # with the second hidden set of nodes
+        self.layer2 = sigmoid(numpy.dot(self.layer1, self.weights2))
+
+        # layer3 is the layer connecting second hidden layer with the output node
+        self.layer3 = sigmoid(numpy.dot(self.layer2, self.weights3))
+
+        return self.layer3
+
+    def back_propagation(self) -> None:
+        """
+        backpropagating between the layers using sigmoid derivative and
+        loss between layers
+        updates the weights between the layers
+
+        >>> input_val =numpy.array(([0,0,0],[0,0,0],[0,0,0]),dtype=float)
+        >>> output_val = numpy.array(([0],[0],[0]),dtype=float)
+        >>> nn = NeuralNetwork(input_val,output_val)
+        >>> res = nn.feedforward()
+        >>> weights = nn.back_propagation()
+        """
+
+        updated_weights3 = numpy.dot(
+            self.layer2.T, 2 * (self.y - self.output) * sigmoid_derivative(self.output)
+        )
+        updated_weights2 = numpy.dot(
+            self.layer1.T,
+            numpy.dot(
+                2 * (self.y - self.output) * sigmoid_derivative(self.output),
+                self.weights3.T,
+            )
+            * sigmoid_derivative(self.layer2),
+        )
+        updated_weights1 = numpy.dot(
+            self.input.T,
+            numpy.dot(
+                numpy.dot(
+                    2 * (self.y - self.output) * sigmoid_derivative(self.output),
+                    self.weights3.T,
+                )
+                * sigmoid_derivative(self.layer2),
+                self.weights2.T,
+            )
+            * sigmoid_derivative(self.layer1),
+        )
+
+        self.weights1 += updated_weights1
+        self.weights2 += updated_weights2
+        self.weights3 += updated_weights3
+
+    def train(self, output: numpy.array, iterations: int, give_loss: bool) -> None:
+        """
+        output : required for calculating loss
+        performs the feeding and back propagation process for the
+        given number of iterations
+        every iteration will update the weights of neural network
+
+        >>> input_val = numpy.array(([0,0,0],[0,1,0],[0,0,1]),dtype=float)
+        >>> output_val = numpy.array(([0],[1],[1]),dtype=float)
+        >>> nn = NeuralNetwork(input_val,output_val)
+        >>> nn.train(output_val,10,False)
+        """
+        for iteration in range(1, iterations + 1):
+            self.output = self.feedforward()
+            self.back_propagation()
+            if give_loss:
+                print(
+                    "Iteration %s " % iteration,
+                    "Loss: "
+                    + str(numpy.mean(numpy.square(output - self.feedforward()))),
+                )
+
+    def predict(self, input: list) -> int:
```

Comment:
Please provide the list elements type: `List[type]` using `from typing import List`